providers/summary/scancode.js

```diff
@@ -28,22 +28,30 @@ class ScanCodeSummarizer {
     const result = {}
     this.addDescribedInfo(result, coordinates, harvested)
     const buckets = this.computeFileBuckets(harvested.content.files, facets)
-    set(result, 'licensed.facets', {})
-    const facetsObject = get(result, 'licensed.facets')
-    const declaredLicenses = new Set()
+    let declaredLicenses = new Set()
+    let packageLicenses = new Set()
+    for (const fileIdx in harvested.content.files) {
+      const file = harvested.content.files[fileIdx]
+      if (!file.path.toLowerCase().includes('/')) {
+        const asserted = get(file, 'packages[0].asserted_licenses')
+        if (asserted && packageLicenses.size === 0) this._addArrayToSet(asserted, packageLicenses, license => license.license || license.spdx_license_key)
+        declaredLicenses = this._summarizeLicenseFile(file)
+        if (declaredLicenses.size > 0) break
```

Comment:
This (`break`) is a bit of smell. Indicates that the code might be doing too much in one function. Suggest here that the declared/package license code be broken out into something like `_summarizeDeclaredInfo`. The other loop might be good to factor out into say `_summarizeDiscoveredInfo`

Reply:
I have broken it up it into 4 helper functions (_summarizeDeclaredLicenseInfo, _summarizePackageInfo, _summarizeFileInfo, and _summarizeDiscoveredLicenseInfo). Please let me know if that is too verbose. I have always been bad at writing modular code, so any comments would be helpful.